panda-market/pages/boards/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { dehydrate, QueryClient } from '@tanstack/react-query';
+
+import { Boards } from '@/pages/boards';
+import { getBoards } from '@/shared/api';
+
+const queryClient = new QueryClient();
+
+export const getServerSideProps = async () => {
+  await queryClient.prefetchQuery({
+    queryKey: ['allBoards'],
+    queryFn: () => getBoards({ page: 1, pageSize: 10, orderBy: 'recent' }),
+  });
```

Comment:
이 부분을 궁금해서 찾아봤는데 서버 요청마다 새로운 쿼리 클라이언트를 생성해야한다고 합니다. 
여러 유저로부터 오는 요청에 대해서 같은 클라이언트를 사용하게되면 캐쉬 데이터가 충돌이 날 수 있습니다. 
SSR 방식으로 데이터를 가져올때는 요청마다 상태를 독립적으로 유지 할 필요가 있습니다. 

```suggestion
export const getServerSideProps = async () => {
const queryClient = new QueryClient();
  await queryClient.prefetchQuery({
    queryKey: ['allBoards'],
    queryFn: () => getBoards({ page: 1, pageSize: 10, orderBy: 'recent' }),
  });
```